### PR TITLE
feat: automatically pick active menu item color by contrast

### LINKED ISSUE #11353: automatically pick menu item text color by comparing contrasts
## What is missing?
The background color for active menu item is picked from customizable color palette, but it's text color is still hardcoded to white´.
Like its already done for buttons, the text color here could also be picked automatically.

## Why is it important?
it makes changing color palette a lot easier, when text colors are picked automatically instead of reconfiguring dozens of variables for.

## Possible Solution
i will send you a PR and refer to this issue

## Context
I customized the color palette and my primary color does not work with default white text color. To fix this issue i had to manually set text color in many variables for menus, dropdown menus, default buttons etc, while text colors for styled buttons are picked automatically

## Your Environment
- Foundation version(s) used: latest
- Browser(s) name and version(s): doesn't matter
- Operating System and version (desktop or mobile): doesn't matter

## Checklist (all required):

- [x] I have read and follow the CONTRIBUTING.md document.
- [x] There are no other issues similar to this one.
- [x] The issue title is descriptive.
- [x] The template is correctly filled.

### PR DESCRIPTION
**Pull request by vanilla-thunder**
picking text color by contrast instead of hardcoded white for easier color scheme customisation

## Description
while active menu item **background** is taken from customizable color palette, it's **color** is still hardcoded to white and needs to be replaced manually to matching color. 

- Closes https://github.com/zurb/foundation-sites/issues/11353

## Motivation and Context
For easier color scheme customization the text color could be picked automatically by comparing color contrasts, like its done for styled buttons ( _buttons.scss line 140 ).

## Types of changes

- [ ] Documentation
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
      functionality to change)  

Imho, this PR is an improvement. Since auto-picking color by contrast is already implemented for styled buttons, i would expect the same behavior for other styled elements. Therefore the issue is kind of "unexpected behavior". I checked both "fix" and "new feature"

## Checklist (all required):

- [x] I have read and follow the [CONTRIBUTING](CONTRIBUTING.md) document.
- [x] There are no other pull request similar to this one.
- [x] The pull request title is descriptive.
- [x] The template is fully and correctly filled.
- [x] The pull request targets the right branch (`develop` or `support/*`).
- [x] My commits are correctly titled and contain all relevant information.
- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly to my changes (if relevant).
- [ ] I have added tests to cover my changes (if relevant).
- [x] All new and existing tests passed.

### CODE REVIEW

File: scss/components/_menu.scss [hunk 1]
```diff
@@ -22,9 +22,13 @@ $menu-items-padding: $global-menu-padding !default;
 /// @type Number
 $menu-simple-margin: 1rem !default;
 
-/// Text color of an active menu item.
+/// Text color for menu items.
 /// @type Color
-$menu-item-color-active: $white !default;
+$menu-item-color: $white !default;
```
Comment (ncoden) [inline scss/components/_menu.scss]: This would be a breaking change

File: scss/components/_menu.scss [hunk 2]
```diff
@@ -22,9 +22,13 @@ $menu-items-padding: $global-menu-padding !default;
...
+
+/// Alternative text color for menu items.
+/// @type Color
+$menu-item-color-alt: $black !default;
```
Comment (ncoden) [inline scss/components/_menu.scss]: I think we should use `$menu-item-color-active` and `$menu-item-color-alt-active` for backward compatibility and concistency.

**Review by ncoden — CHANGES_REQUESTED** [review]
Hi @vanilla-thunder. Thank you so much for your first contribution 🎉 

We try to avoid unecessary renaming, and others things that would break our Sass users code. Please take a loot at the review comments below.

**Comment by ncoden** [comment]
Planned for v6.6.0 as this is a new feature.

**Comment by ncoden** [comment]
@vanilla-thunder Will you have time to discuss/provide the requested changes ?

**Note to myself**: maybe we could simplify the pull request "Checklist" and add a checkbox "I am availaible to discuss and work on this pull request later"

**Comment by vanilla-thunder** [comment]
sure, sorry for letting you wait.

I had some additional thoughts about customizing menu colors, especially the regular menu item text, not only the active. Just in case someone would like to have different menu color than regular text color (e.g. when having dark navbar while the rest of the page is light), you might want to keep the original variable for active color, but also add variables for regular menu item color and alternative color for active menu item. 
Something like this:
````SASS
/// regular menu colors
$menu-item-color: $white !default;
$menu-item-color-alt: $black !default;
/// active item colors
$menu-item-color-active: $menu-item-color !default;
$menu-item-color-alt-active: $menu-item-color-alt !default;
````

So people who already configured own $menu-item-color-active will keep it.
What do you think about that?

Also, i found the same color picking bahavior in dropdown-menus, do you want me to make separate PR for dropdowns or add my changes into this PR?

**Comment by ncoden** [comment]
For now, menus doesn't have a color set for their default state. If I understand your proposal correctly, it would change this color to $white. Is that what you mean ?

For the pull requests, please group all similar changes even across multiple components. It's easier for us to follow. Please update the pull request title and description.

**Comment by vanilla-thunder** [comment]
please have a look at the changes, this should be finde now.
(forget my other suggestion, I accidentally looked into wrong files)

**Review by ncoden — APPROVED** [review]
LGTM. Could you squash everything ?

**Comment by ncoden** [comment]
Thank you

**Pull request merged**